metagpt/provider/bedrock/amazon_bedrock_api.py

```diff
@@ -0,0 +1,136 @@
+        ……
+        ```
+        """
+        client = self.__init_client("bedrock")
+        # only output text-generation models
+        response = client.list_foundation_models(byOutputModality='TEXT')
+        summaries = [f'{summary["modelId"]:50} Support Streaming:{summary["responseStreamingSupported"]}'
+                     for summary in response["modelSummaries"]]
+        logger.info("\n"+"\n".join(summaries))
+
+    def invoke_model(self, request_body) -> dict:
+        response = self.__client.invoke_model(
+            modelId=self.config.model, body=request_body
+        )
+        response_body = self._get_response_body(response)
+        return response_body
+
+    def invoke_model_with_response_stream(self, request_body) -> EventStream:
+        response = self.__client.invoke_model_with_response_stream(
+            modelId=self.config.model, body=request_body
+        )
+        return response
+
+    @property
+    def _generate_kwargs(self) -> dict:
+        model_max_tokens = get_max_tokens(self.config.model)
+        if self.config.max_token > model_max_tokens:
+            max_tokens = model_max_tokens
+        else:
+            max_tokens = self.config.max_token
+
+        return {
+            self.__provider.max_tokens_field_name: max_tokens,
+            "temperature": self.config.temperature
+        }
+
+    def completion(self, messages: list[dict]) -> str:
+        request_body = self.__provider.get_request_body(
+            messages, self._generate_kwargs)
+        response_body = self.invoke_model(request_body)
+        completions = self.__provider.get_choice_text(response_body)
+        return completions
+
+    def _chat_completion_stream(self, messages: list[dict], timeout=USE_CONFIG_TIMEOUT) -> str:
+        if self.config.model in NOT_SUUPORT_STREAM_MODELS:
+            logger.warning(
+                f"model {self.config.model} doesn't support streaming output!")
+            return self.completion(messages)
+
+        request_body = self.__provider.get_request_body(
+            messages, self._generate_kwargs, stream=True)
+
+        response = self.invoke_model_with_response_stream(request_body)
+        collected_content = []
+        for event in response["body"]:
+            chunk_text = self.__provider.get_choice_text_from_stream(event)
+            collected_content.append(chunk_text)
+            log_llm_stream(chunk_text)
+
+        log_llm_stream("\n")
+        full_text = ("".join(collected_content)).lstrip()
+        return full_text
+
+    # boto3 don't support support asynchronous calls.
+    # for asynchronous version of boto3, check out:
+    # https://aioboto3.readthedocs.io/en/latest/usage.html
+    # However,aioboto3 doesn't support invoke model
+
+    async def acompletion(self, messages: list[dict]):
+        return await self._achat_completion(messages)
+
+    async def acompletion_text(self, messages: list[dict], stream: bool = False,
```

Comment:
see acompletion_text in BaseLLM, there has a get_choice_text to extract final response text. Maybe you can put your self.__provider.get_choice_text inside it. So we can keep the whole structure.